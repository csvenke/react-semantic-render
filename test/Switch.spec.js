import React from "react";
import renderer from "react-test-renderer";

import { Switch as Switch1 } from "../dist/index.cjs";
import { Switch as Switch2 } from "../dist/index.esm";
import Switch3 from "../dist/Switch";
import Switch4 from "../src/Switch";

const describeEach = describe.each([Switch1, Switch2, Switch3, Switch4]);

const makeCreateElement = Switch => (value, values, withDefault = true) => (
  <Switch value={value}>
    {values.map((item, index) => (
      <Switch.Case key={item} value={item}>
        <div>{`case ${index}`}</div>
      </Switch.Case>
    ))}
    {withDefault ? (
      <Switch.Default>
        <div>default</div>
      </Switch.Default>
    ) : null}
  </Switch>
);

describeEach("without default case", Switch => {
  const createElement = makeCreateElement(Switch);

  it("should return content from first case", () => {
    const component = renderer.create(createElement(0, [0, 1, 2], false));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content from second case", () => {
    const component = renderer.create(createElement(1, [0, 1, 2], false));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content from third case", () => {
    const component = renderer.create(createElement(2, [0, 1, 2], false));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return null", () => {
    const component = renderer.create(createElement(3, [0, 1, 2], false));
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("with default case", Switch => {
  const createElement = makeCreateElement(Switch);

  it("should return content from first case", () => {
    const component = renderer.create(createElement(0, [0, 1, 2]));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content from second case", () => {
    const component = renderer.create(createElement(1, [0, 1, 2]));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content from third case", () => {
    const component = renderer.create(createElement(2, [0, 1, 2]));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return null", () => {
    const component = renderer.create(createElement(3, [0, 1, 2]));
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("input tests", Switch => {
  const createElement = makeCreateElement(Switch);

  it("should return content when value is number", () => {
    const component = renderer.create(
      createElement(3, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content when value is string", () => {
    const component = renderer.create(
      createElement("salsa", [0, 1, 2, 3, "salsa", 5, 6, 7, 8])
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content when value is boolean (true)", () => {
    const component = renderer.create(
      createElement(true, [0, 1, 2, 3, 4, 5, true, 7, 8, 9])
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should return content when value is boolean (false)", () => {
    const component = renderer.create(
      createElement(false, [0, 1, 2, 3, 4, 5, false, 7, 8, 9])
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("misc tests", Switch => {
  const createElement = makeCreateElement(Switch);

  it("should return content from first occurence when multiple matches", () => {
    const component = renderer.create(
      createElement(true, [0, 1, 2, true, 4, 5, true, 7, 8, 9])
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
