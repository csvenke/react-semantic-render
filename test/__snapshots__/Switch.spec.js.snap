// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input tests should return content when value is boolean (false) 1`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (false) 2`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (false) 3`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (false) 4`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (true) 1`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (true) 2`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (true) 3`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is boolean (true) 4`] = `
<div>
  case 6
</div>
`;

exports[`input tests should return content when value is number 1`] = `
<div>
  case 3
</div>
`;

exports[`input tests should return content when value is number 2`] = `
<div>
  case 3
</div>
`;

exports[`input tests should return content when value is number 3`] = `
<div>
  case 3
</div>
`;

exports[`input tests should return content when value is number 4`] = `
<div>
  case 3
</div>
`;

exports[`input tests should return content when value is string 1`] = `
<div>
  case 4
</div>
`;

exports[`input tests should return content when value is string 2`] = `
<div>
  case 4
</div>
`;

exports[`input tests should return content when value is string 3`] = `
<div>
  case 4
</div>
`;

exports[`input tests should return content when value is string 4`] = `
<div>
  case 4
</div>
`;

exports[`misc tests should return content from first occurence when multiple matches 1`] = `
<div>
  case 3
</div>
`;

exports[`misc tests should return content from first occurence when multiple matches 2`] = `
<div>
  case 3
</div>
`;

exports[`misc tests should return content from first occurence when multiple matches 3`] = `
<div>
  case 3
</div>
`;

exports[`misc tests should return content from first occurence when multiple matches 4`] = `
<div>
  case 3
</div>
`;

exports[`with default case should return content from first case 1`] = `
<div>
  case 0
</div>
`;

exports[`with default case should return content from first case 2`] = `
<div>
  case 0
</div>
`;

exports[`with default case should return content from first case 3`] = `
<div>
  case 0
</div>
`;

exports[`with default case should return content from first case 4`] = `
<div>
  case 0
</div>
`;

exports[`with default case should return content from second case 1`] = `
<div>
  case 1
</div>
`;

exports[`with default case should return content from second case 2`] = `
<div>
  case 1
</div>
`;

exports[`with default case should return content from second case 3`] = `
<div>
  case 1
</div>
`;

exports[`with default case should return content from second case 4`] = `
<div>
  case 1
</div>
`;

exports[`with default case should return content from third case 1`] = `
<div>
  case 2
</div>
`;

exports[`with default case should return content from third case 2`] = `
<div>
  case 2
</div>
`;

exports[`with default case should return content from third case 3`] = `
<div>
  case 2
</div>
`;

exports[`with default case should return content from third case 4`] = `
<div>
  case 2
</div>
`;

exports[`with default case should return null 1`] = `
<div>
  default
</div>
`;

exports[`with default case should return null 2`] = `
<div>
  default
</div>
`;

exports[`with default case should return null 3`] = `
<div>
  default
</div>
`;

exports[`with default case should return null 4`] = `
<div>
  default
</div>
`;

exports[`without default case should return content from first case 1`] = `
<div>
  case 0
</div>
`;

exports[`without default case should return content from first case 2`] = `
<div>
  case 0
</div>
`;

exports[`without default case should return content from first case 3`] = `
<div>
  case 0
</div>
`;

exports[`without default case should return content from first case 4`] = `
<div>
  case 0
</div>
`;

exports[`without default case should return content from second case 1`] = `
<div>
  case 1
</div>
`;

exports[`without default case should return content from second case 2`] = `
<div>
  case 1
</div>
`;

exports[`without default case should return content from second case 3`] = `
<div>
  case 1
</div>
`;

exports[`without default case should return content from second case 4`] = `
<div>
  case 1
</div>
`;

exports[`without default case should return content from third case 1`] = `
<div>
  case 2
</div>
`;

exports[`without default case should return content from third case 2`] = `
<div>
  case 2
</div>
`;

exports[`without default case should return content from third case 3`] = `
<div>
  case 2
</div>
`;

exports[`without default case should return content from third case 4`] = `
<div>
  case 2
</div>
`;

exports[`without default case should return null 1`] = `null`;

exports[`without default case should return null 2`] = `null`;

exports[`without default case should return null 3`] = `null`;

exports[`without default case should return null 4`] = `null`;
